src/query-builder.ts

```diff
@@ -1,161 +1,160 @@
-        return String(obj)
+    if (typeof obj === 'string') {
+      // Check if this is likely an enum (all uppercase with underscores and numbers)
+      if (/^[A-Z0-9_]+$/.test(obj)) {
+        return obj // Return enum without quotes
+      } else {
+        return JSON.stringify(obj) // Return string with quotes
       }
+    }
 
-      if (Array.isArray(obj)) {
-        const items = obj.map((item) => processObj(item)).join(', ')
-        return `[${items}]`
-      }
+    if (typeof obj === 'number' || typeof obj === 'boolean') {
+      return String(obj)
+    }
 
-      if (typeof obj === 'object') {
-        const pairs = Object.entries(obj)
-          .map(([key, value]) => `${key}: ${processObj(value)}`)
-          .join(', ')
-        return `{${pairs}}`
-      }
+    if (Array.isArray(obj)) {
+      const items = obj.map((item) => processObj(item)).join(', ')
+      return `[${items}]`
+    }
 
-      return String(obj)
+    if (typeof obj === 'object') {
+      const pairs = Object.entries(obj)
+        .map(([key, value]) => `${key}: ${processObj(value)}`)
+        .join(', ')
+      return `{${pairs}}`
     }
 
-    return processObj(obj)
+    return String(obj)
   }
 
-  /**
-   * Process a filter for a GraphQL query, handling different filter types appropriately
-   * @param _entityName - The entity being queried (e.g., 'tokens', 'commitments')
-   * @param filterName - The filter name (e.g., 'where', 'orderBy', 'limit')
-   * @param value - The filter value with appropriate type based on FilterValue
-   * @returns Formatted GraphQL argument string
-   */
-  processFilter<K extends keyof QueryIO, F extends keyof QueryIO[K]['input']>(
-    _entityName: K,
-    filterName: F,
-    value: FilterValue<K, F>
-  ): string {
-    switch (filterName) {
-      case 'where':
-        return value ? `where: ${this.jsonToGraphQLArgs(value)}` : ''
-
-      case 'orderBy':
-        if (Array.isArray(value) && value.length > 0) {
-          const orderByValues = value
-            .map(order => String(order).replace(/["']/g, ''))
-            .join(', ')
-          return `orderBy: [${orderByValues}]`
-        }
-        return ''
-
-      // All of these are `number`
-      case 'limit':
-      case 'offset':
-      case 'first':
-        return value !== undefined ? `${String(filterName)}: ${value}` : ''
-      case 'after':
-        return value ? `after: ${JSON.stringify(value)}` : ''
-
-      default:
-        return ''
-    }
+  return processObj(obj)
+}
+
+/**
+ * Process a filter for a GraphQL query, handling different filter types appropriately
+ * @param _entityName - The entity being queried (e.g., 'tokens', 'commitments')
+ * @param filterName - The filter name (e.g., 'where', 'orderBy', 'limit')
+ * @param value - The filter value with appropriate type based on FilterValue
+ * @returns Formatted GraphQL argument string
+ */
+const processFilter = <K extends keyof EntityQueryMap, F extends keyof EntityQueryMap[K]['input']>(
```

Comment:
is there a reason to use arrow functions here? In general its better to use declared functions for better error tracking